### PR TITLE
Make EWMA alpha configurable in FixedExecutor

### PR DESCRIPTION
**Pull request by pxsalehi**
For Stateless autoscaling, we'd need a different Alpha to track the task 
execution time EWMA. This change makes the EWMA Alpha configurable 
and uses a different value for the Write executors.

Closes ES-6325

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java:139]: Instead of adding an alternate signature, could you please update the existing uses of newFixed? It seems like a trivial update to the signature (can be done with intellij refactoring tools), and there are only 21 uses I see (almost all for tests).

File: server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java:411]: nit: the casing on this is difficult to read. Could it be camelCase like `ewmaAlpha`?

File: server/src/main/java/org/elasticsearch/threadpool/FixedExecutorBuilder.java
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/threadpool/FixedExecutorBuilder.java:62]: Similar to my other comment, there are only 19 uses of the larger ctor (taking more params). Can we avoid adding yet-another-ctor by adjusting the more full fledged ctor?

**Review by rjernst — COMMENTED** [review]
I left a couple thoughts

**Comment by pxsalehi** [comment]
Thanks for the review @rjernst. All done in https://github.com/elastic/elasticsearch/pull/97424/commits/9f6086ff49526caf0b6625f73ccdae741677d0e5 (and https://github.com/elastic/elasticsearch/pull/97424/commits/9184412c5d28fa5836b5d2f5522aa65b4e7d1425). Please have another look.

File: server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java
Comment (kingherc) [inline server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java:192]: Here and in other places in the PR, I'm not at all sure what's 0.1 compared to the previous 0.3. Has this been already discussed/agreed? I'm starting to think I may not be the best one to review this finally.

File: server/src/test/java/org/elasticsearch/common/util/concurrent/EsExecutorsTests.java
Comment (kingherc) [inline server/src/test/java/org/elasticsearch/common/util/concurrent/EsExecutorsTests.java:124]: nit Seems like you can refactor this in a function. Or make the DEFAULT/DO_NOT_TRACK enum values and select a random enum value.

File: server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java
Comment (pxsalehi) [reply server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java:192]: The new value is based on some estimate that is mentioned in the attached Jira issue if you are interested. We could change the value to something else later, but for now it seems the default (0.3) might not be ideal. At the end the main change is making it configurable, the value itself is only having some effect on what we consider is the average task length for autoscaling estimates.

Comment (kingherc) [reply server/src/main/java/org/elasticsearch/threadpool/ThreadPool.java:192]: I see thanks! Makes sense by reading it quickly, but I'll leave it to one of the autoscaling colleagues to finally approve the 0.3->0.1 value in these places cc @idegtiarenko . All the rest of the PR LGTM!

File: server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java:398]: nit: match casing of this variable with the class member

File: server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java:402]: nit: could the constant definitions above use the ctor taking both the alpha and boolean? Having two single arg ctors, one public one private, is a little confusing.

**Review by rjernst — APPROVED** [review]
LGTM once the pending issues are resolved. I agree with Iraklis on moving the default alpha constant (maybe rename to note it is the default as well?) to the tracking config class.

**Comment by pxsalehi** [comment]
@elasticmachine run elasticsearch-ci/part-3

(failure was https://github.com/elastic/elasticsearch/issues/97487)

**Comment by pxsalehi** [comment]
@elasticmachine run elasticsearch-ci/packaging-tests-windows-sample

(Infra/Jenkins issue with Windows machines, it seems.)

**Pull request merged**
